compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -2354,6 +2362,84 @@ private STNode parseTypeDescriptorInternal(List<STNode> qualifiers, ParserRuleCo
         }
     }
 
+    private STNode parseQualifiedTypeRefOrTypeDesc(List<STNode> qualifiers, boolean isInConditionalExpr) {
+        STToken preDeclaredPrefix = consume();
+        STToken nextNextToken = getNextNextToken();
+        if (nextNextToken.kind == SyntaxKind.IDENTIFIER_TOKEN) {
+            reportInvalidQualifierList(qualifiers);
+            return parseQualIdentifierWithPredeclPrefix(preDeclaredPrefix, isInConditionalExpr);
+        } else {
```

Comment:
since we are returning from `if`, we don't need the `else` block

Reply:
Done